include/gsl/gsl_assert

```diff
@@ -68,17 +68,32 @@
 
 namespace gsl
 {
+
 struct fail_fast : public std::runtime_error
 {
     explicit fail_fast(char const* const message) : std::runtime_error(message) {}
 };
-}
+
+template <typename Exception>
+void throw_exception(Exception&& exception);
```

Comment:
Given this is implementation detail it should really be in the `gsl::details` namespace (we do that for other helpers around the place).